client/src/components/SignupFormSteps/index.js

```diff
@@ -39,30 +39,28 @@ export default function SignupFormSteps() {
   }, [skipped, activeStep, setActiveStep, setSkipped]);
 
   useEffect(() => {
-    if (Object.keys(data).length === 10) {
-      const { CancelToken } = axios;
-      const source = CancelToken.source();
+    const { CancelToken } = axios;
+    const source = CancelToken.source();
 
-      (async () => {
-        try {
-          setShowLoading(true); // turn the loading progress bar
-          await axios.post('/api/v1/signup', data, {
-            cancelToken: source.token,
-          });
-          setShowLoading(false); // turnoff the loading progress bar after success
-          setErrorMessage('success');
-          return history.push(Login);
-        } catch (err) {
-          setShowLoading(false); // turnoff the loading progress bar after fail
-          return setErrorMessage(
-            err.response.data.message || 'Something went wrong !! '
-          );
-        }
-      })();
+    (async () => {
+      try {
+        setShowLoading(true); // turn the loading progress bar
+        await axios.post('/api/v1/signup', data, {
+          cancelToken: source.token,
+        });
+        setShowLoading(false); // turnoff the loading progress bar after success
+        setErrorMessage('success');
+        return history.push(Login);
+      } catch (err) {
+        setShowLoading(false); // turnoff the loading progress bar after fail
+        return setErrorMessage(
+          err.response.data.message || 'Something went wrong !! '
+        );
+      }
+    })();
```

Comment:
I don't think you ned to wrap this all (lines 45-60) in an immediately-invoked function.  You should be able to remove it and just add an `async` keyword to the useEffect call above (so line 41 becomes `useEffect(async () => {`
That would be a little simpler.